packages/core/chrome/core-chrome-browser-internal/src/nav_controls/nav_controls_service.ts

```diff
@@ -26,20 +26,34 @@ export class NavControlsService {
     const navControlsExtension$ = new BehaviorSubject<ReadonlySet<ChromeNavControl>>(new Set());
     const helpMenuLinks$ = new BehaviorSubject<ChromeHelpMenuLink[]>([]);
 
+    function register(control$: BehaviorSubject<ReadonlySet<ChromeNavControl>>) {
+      return (toAdd: ChromeNavControl) => {
+        control$.next(new Set([...control$.getValue().values(), toAdd]));
+      };
+    }
+
+    function unregister(control$: BehaviorSubject<ReadonlySet<ChromeNavControl>>) {
+      return (toDelete: ChromeNavControl) => {
+        const set = control$.getValue();
+        if (!set.has(toDelete)) return;
+        const clone = new Set(set);
+        clone.delete(toDelete);
+        control$.next(clone);
+      };
+    }
+
     return {
-      // In the future, registration should be moved to the setup phase. This
-      // is not possible until the legacy nav controls are no longer supported.
-      registerLeft: (navControl: ChromeNavControl) =>
-        navControlsLeft$.next(new Set([...navControlsLeft$.value.values(), navControl])),
+      registerLeft: register(navControlsLeft$),
+      unregisterLeft: unregister(navControlsLeft$),
 
-      registerRight: (navControl: ChromeNavControl) =>
-        navControlsRight$.next(new Set([...navControlsRight$.value.values(), navControl])),
+      registerRight: register(navControlsRight$),
+      unregisterRight: unregister(navControlsRight$),
```

Comment:
Needed for the ability to show/hide the help menu for keyboard shortcuts and tying this to the react component's lifecycle